src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3015,8 +3015,11 @@ std::string Creature::GetScriptName() const
 uint32 Creature::GetScriptId() const
 {
     if (CreatureData const* creatureData = GetCreatureData())
-        if (uint32 scriptId = creatureData->ScriptId)
+    {
+        uint32 scriptId = creatureData->ScriptId;
+        if (scriptId && GetEntry() == creatureData->id1)
```

Comment:
Please explain this change

Reply:
From what I could see, through this change, what can be achieved is that if the npc has an updateEntry, then it takes the script that corresponds to it, otherwise, it always takes the script from the npc before the update. That's why the dialogues worked poorly. Initially, I had proposed something else, which was to spawn the npc, but this is not the first one to have this problem. What is not entirely clear to me, if necessary, put a true in the `bool changelevel`, which is the third parameter of the method.
```cpp
bool Creature::UpdateEntry(uint32 Entry, const CreatureData* data, bool changelevel, bool updateAI)
```

Reply:
How else do we solve the problem? Because if we set that value in True, the script is not updated and there is a flaw that can be seen in the videos. I uploaded several videos explaining the difference when applying the changes, but if there's another way to solve the problem, I'd like to know. The method might not be right, it can be changed. The goal is to correct the problem.

Reply:
From the mobile phone, I didn't see that you had sent a suggestion for change. I applied it, I'll try it. I'll see if I get the same result. Thanks.